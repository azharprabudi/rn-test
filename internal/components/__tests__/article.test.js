const React = require('react');
const {Text} = require('react-native');
const Article = require('../article').default;
const {render} = require('react-native-testing-library');

describe('Testing Article Child Component', () => {
  it('should render correctly', () => {
    const onPress = jest.fn();
    const jsonOfComp = render(
      <Article onPress={onPress} title={''} content={''} publishedAt={''} />,
    ).toJSON();

    expect(jsonOfComp).toMatchSnapshot();
  });

  it('props are render correctly', () => {
    const props = {
      title: 'hello',
      content: 'hello',
      publishedAt: 'hello',
      onPress: jest.fn(),
    };

    const {getByTestId} = render(<Article {...props} />);
    expect(getByTestId('article__title').instance.props.children).toBe(
      props.title,
    );
    expect(getByTestId('article__published-at').instance.props.children).toBe(
      props.publishedAt,
    );
    expect(getByTestId('article__content').instance.props.children).toBe(
      props.content,
    );
  });
});
