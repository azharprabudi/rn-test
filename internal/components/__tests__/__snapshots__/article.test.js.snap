// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Article Child Component should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 2,
      "marginBottom": 6,
      "padding": 4,
    }
  }
>
  <Text
    style={
      Object {
        "flexWrap": "wrap",
        "fontSize": 18,
        "fontWeight": "bold",
      }
    }
    testID="article__title"
  >
    
  </Text>
  <Text
    style={
      Object {
        "fontSize": 12,
      }
    }
    testID="article__published-at"
  >
    
  </Text>
  <Text
    style={
      Object {
        "flexWrap": "wrap",
        "fontSize": 14,
        "marginBottom": 4,
        "marginTop": 4,
      }
    }
    testID="article__content"
  >
    
  </Text>
</View>
`;
